src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -0,0 +1,147 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Dashboard.Model;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Dcp;
+using Aspire.Hosting.Lifecycle;
+using Aspire.Hosting.Utils;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Hosting.Dashboard;
+
+internal sealed class DashboardLifecycleHook(IConfiguration configuration,
+                                             IOptions<DcpOptions> dcpOptions,
+                                             ILogger<DistributedApplication> distributedApplicationLogger,
+                                             IDashboardEndpointProvider dashboardEndpointProvider) : IDistributedApplicationLifecycleHook
+{
+    public Task BeforeStartAsync(DistributedApplicationModel model, CancellationToken cancellationToken)
+    {
+        if (model.Resources.SingleOrDefault(r => StringComparers.ResourceName.Equals(r.Name, KnownResourceNames.AspireDashboard)) is { } dashboardResource)
+        {
+            ConfigureAspireDashboardResource(dashboardResource);
+
+            // Make it first in the list
```

Comment:
```suggestion
            // Make the dashboard first in the list so it starts as fast as possible.
```

Reply:
If you want to go crazy with optimization, you could skip this if the dashboard is already first.